src/BenchmarkDotNet/Columns/BaselineAllocationRatioColumn.cs

```diff
@@ -0,0 +1,88 @@
+        public override string ColumnName => "Alloc Ratio";
+
+        public override string GetValue(Summary summary, BenchmarkCase benchmarkCase, Statistics baseline, IReadOnlyDictionary<string, Metric> baselineMetrics,
+            Statistics current, IReadOnlyDictionary<string, Metric> currentMetrics, bool isBaseline)
+        {
+            double? ratio = GetAllocationRatio(currentMetrics, baselineMetrics);
+            double? invertedRatio = GetAllocationRatio(baselineMetrics, currentMetrics);
+
+            if (ratio == null)
+                return "NA";
+
+            var cultureInfo = summary.GetCultureInfo();
+            var ratioStyle = summary?.Style?.RatioStyle ?? RatioStyle.Value;
+
+            bool advancedPrecision = IsNonBaselinesPrecise(summary, baselineMetrics, benchmarkCase);
+            switch (ratioStyle)
+            {
+                case RatioStyle.Value:
+                    return ratio.Value.ToString(advancedPrecision ? "N3" : "N2", cultureInfo);
+                case RatioStyle.Percentage:
+                    return isBaseline
+                        ? ""
+                        : ratio.Value >= 1.0
+                            ? "+" + ((ratio.Value - 1.0) * 100).ToString(advancedPrecision ? "N1" : "N0", cultureInfo) + "%"
+                            : "-" + ((1.0 - ratio.Value) * 100).ToString(advancedPrecision ? "N1" : "N0", cultureInfo) + "%";
+                case RatioStyle.Trend:
+                    return isBaseline
+                        ? ""
+                        : ratio.Value >= 1.0
+                            ? ratio.Value.ToString(advancedPrecision ? "N3" : "N2", cultureInfo) + "x more"
+                            : invertedRatio == null
+                                ? "NA"
+                                : invertedRatio.Value.ToString(advancedPrecision ? "N3" : "N2", cultureInfo) + "x less";
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(summary), ratioStyle, "RatioStyle is not supported");
+            }
+        }
+
+        private static bool IsNonBaselinesPrecise(Summary summary, IReadOnlyDictionary<string, Metric> baselineMetric, BenchmarkCase benchmarkCase)
+        {
+            string logicalGroupKey = summary.GetLogicalGroupKey(benchmarkCase);
+            var nonBaselines = summary.GetNonBaselines(logicalGroupKey);
+            return nonBaselines.Any(c => GetAllocationRatio(summary[c].Metrics, baselineMetric) is > 0 and < 0.01);
+        }
+
+        private static double? GetAllocationRatio(
+            [CanBeNull] IReadOnlyDictionary<string, Metric> current,
+            [CanBeNull] IReadOnlyDictionary<string, Metric> baseline)
+        {
+            double? currentBytes = GetAllocatedBytes(current);
+            double? baselineBytes = GetAllocatedBytes(baseline);
+
+            if (currentBytes == null || baselineBytes == null)
+                return null;
+
+            if (baselineBytes == 0)
```

Comment:
thanks for improving the code and avoiding the exception rather than catching it!

Reply:
the `double?` math is [safe](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7AHwAEAmARgFgAoaosgTgAoAiAAgAYB6dkVtgalZ4YAd1YATCDmAAbGAH5G7AJStOHZQG5aDFnC48+rQY2FjJ0uYv2r1KrTqZsy3XgKGiJU2QsZlbGtpUdE4cnGRuxh7m3lZKAf5BISysAJZ4AGauRoJmXpa+Fj4AdAAKEADOqRipAG4wAJKZ6dUAngEqQdRAA=).

The logic hides special symbols like `inf`, `NaN` for NA for consistency with `Ratio`.

The null checking can be removed too. But it's easy to forget that division for `null` returns `null`.